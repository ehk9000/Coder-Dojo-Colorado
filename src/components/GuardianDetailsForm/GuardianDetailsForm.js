import React from 'react';

function GuardianDetailsForm(props) {

  return (
    <div>
      HI
    </div>
  );
}

export default GuardianDetailsForm;

