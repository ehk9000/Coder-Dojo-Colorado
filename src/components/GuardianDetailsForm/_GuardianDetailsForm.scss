form.GuardianDetailsForm {
  margin: 0 auto;
  width: 50%;
  @include flex-column;

  div.address-section, label {
    @include flex-column;
  }
}