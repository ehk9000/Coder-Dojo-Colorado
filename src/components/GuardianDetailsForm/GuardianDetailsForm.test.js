import React from 'react';
import { shallow } from 'enzyme';
import GuardianDetailsForm from './GuardianDetailsForm';

describe('GuardianDetailsForm', () => {
  let wrapper;
  const mockHandleSubmit = jest.fn();
  const mockHandleChange = jest.fn();
  const mockInputChange = {
    target: {
      name: 'email',
      value: 'nim@sum.com'
    }
  }

  beforeEach(() => {
    wrapper = shallow(
      <GuardianDetailsForm
        handleSubmit={mockHandleSubmit}
        handleChange={mockHandleChange}
      />
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should send name and value to handleChange props on input change', () => {
    wrapper.find('input#email-input').simulate('change', mockInputChange);
    const { name, value } = mockInputChange.target;
    expect(mockHandleChange).toHaveBeenCalledWith(name, value);
  });

  it('should invoke handleSubmit prop on form submit', () => {
    wrapper.find('form.GuardianDetailsForm').simulate('submit');
    expect(mockHandleSubmit).toHaveBeenCalledTimes(1);
  });
});
