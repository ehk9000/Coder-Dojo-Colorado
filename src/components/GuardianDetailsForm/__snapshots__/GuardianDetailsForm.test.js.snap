// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuardianDetailsForm should match snapshot 1`] = `
<form
  className="GuardianDetailsForm"
  onSubmit={[Function]}
>
  <h2>
    Guardian Details
  </h2>
  <label
    htmlFor="email-input"
  >
    Email:
    <input
      id="email-input"
      name="email"
      onChange={[Function]}
      type="email"
    />
  </label>
  <label
    htmlFor="phoneNumber-input"
  >
    Phone Number:
    <input
      id="phoneNumber-input"
      name="phoneNumber"
      onChange={[Function]}
      type="text"
    />
  </label>
  <div
    className="adress-section"
  >
    <label
      htmlFor="street1-input"
    >
      Street 1:
      <input
        id="street1-input"
        name="street1"
        onChange={[Function]}
        type="text"
      />
    </label>
    <label
      htmlFor="street2-input"
    >
      Street 2:
      <input
        id="street2-input"
        name="street2"
        onChange={[Function]}
        type="text"
      />
    </label>
    <label
      htmlFor="city-input"
    >
      City:
      <input
        id="city-input"
        name="city"
        onChange={[Function]}
        type="text"
      />
    </label>
    <label
      htmlFor="state-input"
    >
      State:
      <input
        id="state-input"
        name="state"
        onChange={[Function]}
        type="text"
      />
    </label>
    <label
      htmlFor="zip-input"
    >
      Zip Code:
      <input
        id="zip-input"
        name="zip"
        onChange={[Function]}
        type="text"
      />
    </label>
  </div>
  <button
    className="signin-btn"
    type="submit"
  >
    Submit
  </button>
</form>
`;
